torch/csrc/inductor/aoti_eager/kernel_holder.cpp

```diff
@@ -0,0 +1,283 @@
+#if !defined(C10_MOBILE) && !defined(ANDROID)
+#include <torch/csrc/inductor/aoti_eager/kernel_holder.h>
+
+#include <ATen/ATen.h>
+
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <torch/csrc/PyInterpreter.h>
+#include <torch/csrc/autograd/python_variable.h>
+#include <torch/csrc/inductor/aoti_runner/model_container_runner_cpu.h>
+#ifdef USE_CUDA
+#include <torch/csrc/inductor/aoti_runner/model_container_runner_cuda.h>
+#endif
+#include <torch/csrc/jit/frontend/function_schema_parser.h>
+
+namespace torch::inductor {
+
+namespace {
+enum class IValueType : uint8_t {
+  Tensor,
+  TensorList,
+  OptionalTensorList,
+  Scalar,
+  Invalid,
+};
+
+static bool HandleTensor(
+    const c10::IValue& ivalue,
+    const c10::Device& device,
+    std::vector<at::Tensor>& inputs) {
+  inputs.push_back(ivalue.toTensor());
+  return true;
+}
+
+static bool HandleTensorList(
+    const c10::IValue& ivalue,
+    const c10::Device& device,
+    std::vector<at::Tensor>& inputs) {
+  for (const auto& item : ivalue.toListRef()) {
+    if (!item.isNone()) {
```

Comment:
Is ignoring none correct here?  Doesn't that change the args?

Reply:
I refined the PR. Regarding the `OptionalTensorList`, we will check if the `std::optional<at::Tensor>` has value.